client/src/main/scala/cakesolutions/kafka/KafkaProducer.scala

```diff
@@ -149,7 +151,13 @@ final class KafkaProducer[K, V](val producer: JKafkaProducer[K, V]) {
     */
   def send(record: ProducerRecord[K, V]): Future[RecordMetadata] = {
     val promise = Promise[RecordMetadata]()
-    producer.send(record, producerCallback(promise))
+    Try {
```

Comment:
When you catch the exception in the try-catch expression, you can use the `scala.util.control.NonFatal` to catch all the non-fatal exceptions. E.g.:

```scala
import scala.util.control.NonFatal

try {
  ...
} catch {
  case NonFatal(e) => promise.failure(e)
}
```